### PR TITLE
Build: Use GitHub Actions for CI/CD

### PR DESCRIPTION
**Pull request by keithamus**
This uses [GitHub Actions](https://github.com/features/actions) with the new [CI/CD features](https://github.blog/2019-08-08-github-actions-now-supports-ci-cd/) to test and deploy chai-http to npm

It attempts to mimic the existing travis.yml as closely as possible.

To clarify where the tokens exist; they're in the settings page for this repo under "Security Tokens". `NPM_TOKEN` exists in there, but `github_token` is managed automatically by GitHub Actions.

### CODE REVIEW

**Comment by austince** [comment]
I like the idea of migrating from travis to here! Should we merge #252 and then update this PR to run semantic-release instead of the publish-npm stage?

**Comment by keithamus** [comment]
@austince yes let's merge #252 first.

**Comment by keithamus** [comment]
This passes the Actions CI. Travis is failing as it is trying to support older versions of Node, but it also passes the more recent builds.

I'd say this is good to go, maybe with a ✅ from @chaijs/core?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          - 8 # to be removed 2020-01
+          - 10 # to be removed 2021-01
+          - x.x.x # safety net; don't remove
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node_version }}
+      - run: npm run env
+      - run: npm i -g npm@6 && npm ci
```
Comment (austince) [inline .github/workflows/ci.yml]:
```suggestion
      - run: npm ci
```

If we're dropping support for < 8, npm >= 6 should be installed by default.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12
+          registry-url: https://registry.npmjs.org/
+      - run: npm run release
+        if: github.event == 'push'
+        env:
+          NPM_TOKEN: ${{secrets.NPM_TOKEN}}
```
Comment (austince) [inline .github/workflows/ci.yml]:
```suggestion
          NPM_TOKEN: ${{secrets.NPM_TOKEN}}
          GITHUB_TOKEN: ${{secrets.github_token}}
```
We'll also need the `GITHUB_TOKEN` for semantic-release, and set up rules to allow it to push a chore commit with the release. See: https://github.com/semantic-release/git#git-authentication

File: .github/workflows/ci.yml (hunk 1 above)
Comment (austince) [reply .github/workflows/ci.yml]: Oops, wrong - 5 is default on 8.0.0.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (austince) [reply .github/workflows/ci.yml]: Is there a chaijs bot we could use for this perhaps?

Comment (keithamus) [reply .github/workflows/ci.yml]: The github token is handled transparently as part of Actions

**Comment by austince** [comment]
@chaijs/core sorry to ping you all, but does anyone have a chance to take a look at this? There's continued interest in adding PRs (https://github.com/chaijs/chai-http/issues/259) and just hoping we can get this in before blocking any of them. Thanks!

**Comment by keithamus** [comment]
I'll just go ahead and merge this one 😆

**Pull request merged**